crates/modelardb_server/src/main.rs

```diff
@@ -103,23 +103,29 @@ fn main() -> Result<(), String> {
     let stdout_log = tracing_subscriber::fmt::layer();
     tracing_subscriber::registry().with(stdout_log).init();
 
-    let arguments = collect_command_line_arguments(3);
-    let arguments: Vec<&str> = arguments.iter().map(|arg| arg.as_str()).collect();
-    let (server_mode, data_folders) = parse_command_line_arguments(&arguments)?;
-
     // Create a Tokio runtime for executing asynchronous tasks. The runtime is
     // not in the context so it can be passed to the components in the context.
     let runtime = Arc::new(
         Runtime::new().map_err(|error| format!("Unable to create a Tokio Runtime: {error}"))?,
     );
 
-    // If a remote data folder was provided, check that it can be accessed. This check is performed
-    // after parse_command_line_arguments() as the Tokio Runtime is required.
-    if let Some(remote_data_folder) = &data_folders.remote_data_folder {
-        runtime.block_on(async {
-            validate_remote_data_folder_from_argument(arguments.get(2).unwrap(), remote_data_folder)
+    let arguments = collect_command_line_arguments(3);
+    let arguments: Vec<&str> = arguments.iter().map(|arg| arg.as_str()).collect();
+    let (server_mode, cluster_mode, data_folders) =
+        runtime.block_on(parse_command_line_arguments(&arguments))?;
+
+    // If a remote data folder was provided, check that it can be accessed. If the cluster mode is
+    // "MultiNode" we assume the remote object store was validated by the manager.
```

Comment:
Probably best to always check the remote object store connection to make it easier to debug problems where the network configuration allows the manager to connect to the object store but not the node as they may run on different machines.

Reply:
The issue is that we need to remote data folder type which we normally get from the argument but in the case of MultiNode we would have to get from the manager. It is included in the bytes we it would mean that we would have to change the function to parse the command line arguments again and make it even more complex.

Ideally we can remove the need for having the remote data folder type when validating so I have added a TODO to look into this again.